File: 3. DSA/02. C++/03. Functions/Question01.cpp
// Add Digits

#include <iostream>
using namespace std;

int addDigits(int);

int main()
{
    int num;
    cout << "Enter number: ";
    cin >> num;

    int answer = addDigits(num);
    cout << "\nSum of digits: " << answer << endl;
}

int addDigits(int num) 
{
        int result = 0;

        while(num)
        {
            result = result + num % 10;
            num /= 10;

            if(num == 0 && result >= 10)
            {
                num = result;
                result = 0;
            }
        }

        return result;
    }

File: 3. DSA/04. Python/03. Functions/Question01.py
# Add Digits

class Solution:
    def addDigits(self, num: int) -> int:
        result = 0
        while(num != 0):
            result += num % 10
            num //= 10

            if(num == 0 and result >= 10):
                num = result
                result = 0
        
        return result
    
num = int(input("Enter number: "))

obj = Solution()
answer = obj.addDigits(num)

print(f"Sum of digits: {answer}")